web/source/dom/preProcessor.ts

```diff
@@ -0,0 +1,278 @@
+namespace com.keyman.dom {
+  export class PreProcessor {
+    /**
+     * Function     _GetEventKeyCode
+     * Scope        Private
+     * @param       {Event}       e         Event object
+     * Description  Finds the key code represented by the event.
+     */
+    static _GetEventKeyCode(e: KeyboardEvent) {
+      if (e.keyCode) {
+        return e.keyCode;
+      } else if (e.which) {
+        return e.which;
+      } else {
+        return null;
+      }
+    }
+
+    /**
+     * Function     _GetKeyEventProperties
+     * Scope        Private
+     * @param       {Event}       e         Event object
+     * @param       {boolean=}    keyState  true if call results from a keyDown event, false if keyUp, undefined if keyPress
+     * @return      {Object.<string,*>}     KMW keyboard event object: 
+     * Description  Get object with target element, key code, shift state, virtual key state 
+     *                Ltarg=target element
+     *                Lcode=keyCode
+     *                Lmodifiers=shiftState
+     *                LisVirtualKeyCode e.g. ctrl/alt key
+     *                LisVirtualKey     e.g. Virtual key or non-keypress event
+     */    
+    static _GetKeyEventProperties(e: KeyboardEvent, keyState?: boolean): text.KeyEvent {
+      let keyman = com.keyman.singleton;
+      let processor = keyman.textProcessor;
+      var s = new text.KeyEvent();
+
+      e = keyman._GetEventObject(e);   // I2404 - Manage IE events in IFRAMEs
```

Comment:
I take it this will be resolved in a future PR? (Also may not be required now given it's IE-focused?)

Reply:
?

I believe that old function exists to solve the issue listed in the comment.  Also, this part will remain on the DOM side... so... what needs to be resolved here?

Reply:
> Also, this part will remain on the DOM side...

I see. 

However, if it's IE only, then we can drop it, right?

Reply:
I mean, we can... but I see no need to explicitly break IE at the moment.  I'd rather leave that for whenever we bother to remove the actual method; there are about 10 other references to it.